angular/src/directives/control-value-accessors/value-accessor.ts

```diff
@@ -1,11 +1,29 @@
-import { AfterViewInit, ElementRef, Injector, OnDestroy, Directive, HostListener } from '@angular/core';
-import { ControlValueAccessor, NgControl } from '@angular/forms';
+import {
+  AfterViewInit,
+  ElementRef,
+  Injector,
+  OnDestroy,
+  Directive,
+  HostListener,
+  Input,
+  forwardRef,
+  Provider,
+} from '@angular/core';
+import { ControlValueAccessor, NgControl, NG_VALUE_ACCESSOR } from '@angular/forms';
 import { Subscription } from 'rxjs';
 
 import { raf } from '../../util/util';
 
+export const valueAccessorProvider = (token: any): Provider => ({
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => token),
+  multi: true,
+});
+
 @Directive()
 export class ValueAccessor implements ControlValueAccessor, AfterViewInit, OnDestroy {
+  @Input() ngModel: any;
```

Comment:
If we wanted to keep our base class clean of this moneky-patch, we could create new directives with selectors for `ngModel` vs. `formControl/formControlName`. It would create x2 our value accessor files, but lowers nested complexity.